lib/src/secure-route.tsx

```diff
@@ -31,15 +31,28 @@ import { SecureRouteInterface } from "./models";
 const SecureRoute: FunctionComponent<SecureRouteInterface & RouteProps> = (props: SecureRouteInterface & RouteProps
 ): ReactElement => {
     const { state } = useAuthContext();
+    const reRenderCheckRef: MutableRefObject<boolean> = useRef(false);
 
     const { component: Component, callback, ...rest } = props;
 
     useEffect(() => {
-        if (!state.isAuthenticated) {
-            callback && callback();
+        if (state.isAuthenticated || state.isLoading) {
+            return;
         }
 
-    }, [ state.isAuthenticated ]);
+        // React 18.x Strict.Mode has a new check for `Ensuring reusable state` to facilitate an upcoming react feature.
```

Comment:
Shall we using block comment style here?

Reply:
For inner function comments, the usual JS Doc approach is to use multiline comments.

Ref: https://github.com/facebook/react/blob/main/packages/react-dom/src/client/ReactDOMSelection.js#L26